torch/distributions.py

```diff
@@ -168,14 +174,20 @@ class Normal(Distribution):
     Args:
         mean (float or Tensor or Variable): mean of the distribution
         std (float or Tensor or Variable): standard deviation of the distribution
+        reparameterized (bool): whether to use reparameterization trick
     """
 
-    def __init__(self, mean, std):
+    def __init__(self, mean, std, reparameterized = False):
         self.mean = mean
         self.std = std
+        super(Normal, self).__init__(reparameterized=reparameterized)
 
     def sample(self):
-        return torch.normal(self.mean, self.std)
+        if self.reparameterized:
```

Comment:
As I understand `.reparameterized` is merely an "under construction" marker that says "if you call `your_dist.sample(...)` with `Variable`s that require gradients, then you'll get a `NotImplementedError`". It looks like you're using it as an alternative to `mean.requires_grad` and `std.requires_grad`. I think this conditional should simply be:
```py
if self.mean.requires_grad or self.std.requires_grad:
    eps = torch.normal(torch.zeros_like(self.mean), torch.ones_like(self.std))
    return self.mean + self.std * eps
else:
    return torch.normal(self.mean, self.std)
```
What do you think?